src/lib/FreeC/Backend/Agda/Converter/TypeDecl.hs

```diff
@@ -0,0 +1,110 @@
+-- | This module contains functions for converting type synonym and data type
+--   declarations and their constructors.
+
+module FreeC.Backend.Agda.Converter.TypeDecl
+  ( convertTypeDecl
+  )
+where
+
+import           Data.List.Extra                ( snoc )
+
+import qualified FreeC.IR.Syntax               as IR
+import           FreeC.IR.SrcSpan               ( SrcSpan(NoSrcSpan) )
+import qualified FreeC.Backend.Agda.Syntax     as Agda
+import qualified FreeC.Backend.Agda.Base       as Agda.Base
+import           FreeC.Backend.Agda.Converter.Free
+import           FreeC.Backend.Agda.Converter.Type
+                                                ( convertConstructorType )
+import           FreeC.Backend.Agda.Converter.Arg
+                                                ( convertTypeVarDecl )
+import           FreeC.Environment.Fresh        ( freshAgdaVar )
+import           FreeC.Environment.LookupOrFail
+import           FreeC.Monad.Converter          ( Converter
+                                                , localEnv
+                                                )
+
+-- | Converts a data or type synonym declaration.
```

Comment:
Maybe the comment should mention, that the current state does not handle type synonyms (even if it's only temporary due to vertical prototyping)?